### ISSUE TITLE
The Flatpak version of element only shows the /home directory when clicking "save" button after downloading a file

### ISSUE THREAD

**Issue by gmanskibiditoilet** [flathub/im.riot.Riot#492]
When i tried to download a file it only showed the home directory as well as other directories. Is that a flatpak issue or does it come from element itself?

I want to the application to show the full home directory including downloads, desktops, etc....

We need the devs to fix asap

https://github.com/user-attachments/assets/55881349-1a70-469d-bb82-905709f0af99

**Comment by Erven2016** [comment]
Because element flatpak is running in sandbox, and user's home can not be accessed defaultly.

You can use [flatseal](https://flathub.org/apps/com.github.tchx84.Flatseal) to grant filesystem acceesses for element:
![Screenshot From 2024-11-26 05-36-00](https://github.com/user-attachments/assets/91acb567-7efb-45d7-9254-91f582d927e9)

Or running [flatpak override](https://docs.flatpak.org/en/latest/flatpak-command-reference.html#idm4018) commands, for example:

```bash
# exposing ~/Downloads to element flatpak
flatpak override --filesystem=xdg-download im.riot.Riot
```

After modification, you need to restart element then be effective.

However, It is not suggested to expose full home directory to a sandbox application for private and avioding pollution of home directory.

**Comment by A6GibKm** [comment]
This is not the problem. The problem is https://github.com/electron/electron/issues/43819.

**Comment by Erven2016** [comment]
> This is not the problem. The problem is [electron/electron#43819](https://github.com/electron/electron/issues/43819).

:scream: 

So actually, element can access files through file picker via `xdg-desktop-portal` before. Even the function of `drag and drop` was working???

I'm sorry about I don't kown this point.

**Comment by A6GibKm** [comment]
Yes, everything was working until a few weeks ago. Newer electron versions have this fixed and it is a matter of time until Electron upgrades to one such version.

**Comment by gmanskibiditoilet** [comment]
hope electron will fix this issue asap

**Comment by gmanskibiditoilet** [comment]
i think the best way for element to fetch /home directory as of right now is to enable file access via a third party application called flatseal
https://flathub.org/apps/com.github.tchx84.Flatseal
